File: src/ios/model/NetworkQueueModel.swift
import Foundation

public class NetworkQueueModel { 

    private var msgId: String
    private  var type: String
    private var priority: Int64
    private  var eventCount: Int64
    private var timestamp: Int64
    public var request: [String: Any]
    private var config: [String: Any]?
    private var failedCount = 0

    init(_ msgId: String, _ type: String, _ priority: Int64, _ timestamp: Int64, _ config: [String: Any], _ eventCount: Int64, _ request: [String: Any], _ failedCount : Int) {
        self.msgId = msgId
        self.type = type
        self.priority = priority
        self.timestamp = timestamp
        self.request = request
        self.config = config
        self.eventCount = eventCount
        self.failedCount = failedCount
    }

    public func getId() -> String {
        return self.msgId
    }
    
    public func getFailedCount() -> Int {
        return self.failedCount
    }
    
    public func setFailedCount(_ count: Int) {
        self.failedCount = count
    }

    public func getType() -> String {
        return self.type
    }

    public func getPriority() -> Int64 {
        return self.priority
    }

    public func getTimestamp() -> Int64{
        return self.timestamp
    }

    public func getRequest() -> [String: Any] {
        return self.request
    }

    public func getEventCount() -> Int64 {
        return self.eventCount
    }

    public func getConfig() -> [String: Any]? {
        if let config = self.config {
            return config
        }
        return nil
    }
}

File: src/android/org/sunbird/sync/model/NetworkQueueModel.java
package org.sunbird.sync.model;

import androidx.annotation.NonNull;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by swayangjit on 25/3/20.
 */
public class NetworkQueueModel implements Comparable<NetworkQueueModel>{
    private String msgId;
    private String type;
    private Integer priority;
    private Integer eventCount;
    private Long timestamp;
    private Request request;
    private String config;

    public NetworkQueueModel(String msgId, String type, Integer priority, Long timestamp, String config, Integer eventCount, Request request) {
        this.msgId = msgId;
        this.type = type;
        this.priority = priority;
        this.timestamp = timestamp;
        this.request = request;
        this.config = config;
        this.eventCount = eventCount;
    }

    public String getId() {
        return msgId;
    }

    public String getType() {
        return type;
    }

    public Integer getPriority() {
        return priority;
    }

    public Long getTimestamp() {
        return timestamp;
    }

    public Request getRequest() {
        return request;
    }

    public Integer getEventCount() {
        return eventCount;
    }

    public JSONObject getConfig() throws JSONException {
        if(config != null){
            return new JSONObject(config);
        }
        return null;
    }

    @Override
    public int compareTo(NetworkQueueModel networkQueueModel) {
        int currentFailedSyncCount = this.getRequest().getNoOfFailureSync();
        int failedSyncCount = networkQueueModel.getRequest().getNoOfFailureSync();

        int currentPriority = this.getPriority() + currentFailedSyncCount;
        int priority = networkQueueModel.getPriority() + failedSyncCount;

        if (priority < currentPriority) {
            return 1;
        } else if (priority > currentPriority) {
            return -1;
        }

        return 0;
    }

    @Override
    public String toString() {
        return "NetworkQueueModel{" +
                "msgId='" + msgId + '\'' +
                '}';
    }
}
